sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -537,9 +537,48 @@ class Dataset[T] private[sql](
    */
   @Experimental
   @InterfaceStability.Evolving
-  def checkpoint(eager: Boolean): Dataset[T] = {
```

Comment:
Could you check the test case of `def checkpoint`? At least we need to add a test case.

Reply:
I can try to create a test to localCheckpoint based on the one for checkpoint, but I'm not very familiar with Scala and the Spark scala API, so currently I don't feel at ease to create a meaningful test. Would anybody be up to add one?

Reply:
So we already test checkpoint in DatasetSuite